internal/kafka/internal/workers/cluster_mgrs/cleanup_cluster_external_resources_mgr.go

```diff
@@ -0,0 +1,113 @@
+package cluster_mgrs
+
+import (
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupClusterExternalResourcesWorkerType = "cleanup_cluster_external_resources"
+)
+
+// CleanupClusterExternalResourcesManager represents a worker that periodically reconciles data plane clusters in cleanup state.
+type CleanupClusterExternalResourcesManager struct {
```

Comment:
I am thinking about the name of `CleanupClusterExternalResourcesManager` will a cluster in `cleanup` state be only limited to cleaning up of external cluster resources or could it do other cleanup actions? the name is quite specific and it matches the current behavior but will it match potential future behavior? what if for example as part of cleanup it needs to send some notification, event...?

Reply:
Thinking about it the name also diverges a bit from the pattern `state_mgr.go`. In that regard, I am thinking of naming the file to just `cleanup_cluster_mgr.go` and the type to `CleanupClusterManager`, what do you think?

Reply:
The proposed name in plural would be good.
The reason for plural is to be consistent with the kafka managers

Reply:
Perfect, thanks. Changing now

Reply:
Change pushed